dagshub/data_engine/model/datasource.py

```diff
@@ -344,8 +372,11 @@ def func():
             self.source.get_from_dagshub()
             send_analytics_event("Client_DataEngine_addEnrichments", repo=self.source.repoApi)
             ctx = MetadataContextManager(self)
+
+            ctx.start_explicit_ds_ctx()
```

Comment:
I think it's better to just store it in another field in the datasource object

```
ds._explicit_update_context = ctx
```

And you set it to None after you're done with it.

Reply:
That'd skip having to do the indirection with the counter